x-pack/plugins/apm/server/lib/service_map/get_max_anomalies.ts

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { Logger } from 'kibana/server';
+import { Setup, SetupTimeRange } from '../helpers/setup_request';
+import { PromiseReturnType } from '../../../typings/common';
+import { ML_GROUP_NAME_APM } from '../anomaly_detection/create_anomaly_detection_jobs';
+import {
+  TRANSACTION_PAGE_LOAD,
+  TRANSACTION_REQUEST,
+} from '../../../common/transaction_types';
+import {
+  TRANSACTION_TYPE,
+  SERVICE_NAME,
+} from '../../../common/elasticsearch_fieldnames';
+import { MaxAnomaly } from '../../../common/anomaly_detection';
+
+const DEFAULT_VALUE = {
+  mlJobIds: [],
+  maxAnomalies: [],
+};
+
+export type MaxAnomaliesResponse = PromiseReturnType<typeof getMaxAnomalies>;
+
+export async function getMaxAnomalies({
+  setup,
+  logger,
+  environment,
+}: {
+  setup: Setup & SetupTimeRange;
+  logger: Logger;
+  environment?: string;
+}) {
+  const { ml, start, end } = setup;
+
+  if (!ml) {
+    logger.warn('Anomaly detection plugin is not available.');
+    return DEFAULT_VALUE;
+  }
+  const mlCapabilities = await ml.mlSystem.mlCapabilities();
+  if (!mlCapabilities.mlFeatureEnabledInSpace) {
+    logger.warn('Anomaly detection feature is not enabled for the space.');
+    return DEFAULT_VALUE;
+  }
+  if (!mlCapabilities.isPlatinumOrTrialLicense) {
+    logger.warn(
+      'Unable to create anomaly detection jobs due to insufficient license.'
+    );
+    return DEFAULT_VALUE;
+  }
+
+  let mlJobIds: string[] = [];
+  try {
+    mlJobIds = await getMLJobIds(ml, environment);
+  } catch (error) {
+    logger.error(error);
+    return DEFAULT_VALUE;
+  }
+
+  const params = {
+    body: {
+      size: 0,
+      query: {
+        bool: {
+          filter: [
+            {
+              term: {
+                result_type: 'record',
+              },
+            },
+            {
+              terms: {
+                job_id: mlJobIds,
+              },
+            },
+            {
+              range: {
+                timestamp: {
+                  gte: start,
+                  lte: end,
+                  format: 'epoch_millis',
+                },
+              },
+            },
+            {
+              terms: {
+                by_field_value: [TRANSACTION_REQUEST, TRANSACTION_PAGE_LOAD],
+              },
+            },
+          ],
+        },
+      },
+      aggs: {
+        service_name: {
```

Comment:
Shouldn't this be plural since it returns a list of buckets?
```suggestion
        services: {
```